src/items/traits.md

```diff
@@ -62,12 +62,12 @@ trait Seq<T> {
 }
 ```
 
-## Object Safety
+## Trait-Object Safety <a id="object-safety"></a>
```

Comment:
```suggestion
## Dyn Compatibility
```

NB: I'd love the URL fragment (named anchor) to get changed to `dyn-compatibility` and have an HTTP redirect from `object-safety`. Although I haven't checked if the Reference is set up to support redirects. I know that the rustc-dev-guide is.

Reply:
We should add a paragraph or footnote pointing out that this concept is "formerly known as object safety".

Reply:
> NB: I'd love the URL fragment (named anchor) to get changed to dyn-compatibility and have an HTTP redirect from object-safety

This would be quite nice because we show this URL literally in rustc's diagnostics. As a result of that we currently have a mismatch of terminology (diagnostic text vs. URL). Of course, that's expected but I'd like this to be temporary only (namely until this PR hits stable (rustc's diagnostics usually link to the stable channel of the Reference)).

Reply:
When renaming sections, you need to use a `<script>` tag to redirect the fragment, see https://github.com/rust-lang/reference/blob/c64e52a3d306eac0129f3ad6c6d8806ab99ae2e9/src/expressions/operator-expr.md?plain=1#L687-L699 as an example.